xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp

```diff
@@ -162,149 +131,26 @@ void CAEBitstreamPacker::Reset()
   m_packedBuffer[0] = 0;
 }
 
-/* we need to pack 24 TrueHD audio units into the unknown MAT format before packing into IEC61937 */
-void CAEBitstreamPacker::PackTrueHD(CAEStreamInfo &info, uint8_t* data, int size)
+void CAEBitstreamPacker::GetDataTrueHD()
 {
-  /* create the buffer if it doesn't already exist */
-  if (m_trueHD[0].empty())
+  // limits a bit MAT frames output speed as this is called every 1/1200 seconds and
+  // anyway only is possible obtain a MAT frame every 12 audio units (TrueHD has 24 units
+  // but are send to packer every 12 to reduce latency). As MAT packer can generate more than
+  // one MAT frame at time (but in average only one every 20 ms) this delays the output
+  // a little when thera are more that one frame at queue but still allows the queue to empty.
+  if (m_dataCountTrueHD > 0)
   {
-    m_trueHD[0].resize(MAT_FRAME_SIZE);
-    m_trueHD[1].resize(MAT_FRAME_SIZE);
-    m_thd = {};
-  }
-
-  if (size < 10)
+    m_dataCountTrueHD--;
     return;
-
-  uint8_t* pBuf = m_trueHD[m_thd.bufferIndex].data();
-  const uint8_t* pData = data;
-
-  int totalFrameSize = size;
-  int dataRem = size;
-  int paddingRem = 0;
-  int ratebits = 0;
-  int nextCodeIdx = 0;
-  uint16_t inputTiming = 0;
-  bool havePacket = false;
-
-  if (AV_RB24(data + 4) == 0xf8726f)
-  {
-    /* major sync unit, fetch sample rate */
-    if (data[7] == 0xba)
-      ratebits = data[8] >> 4;
-    else if (data[7] == 0xbb)
-      ratebits = data[9] >> 4;
-    else
-      return;
-
-    m_thd.samplesPerFrame = 40 << (ratebits & 3);
   }
 
-  if (!m_thd.samplesPerFrame)
-    return;
-
-  inputTiming = AV_RB16(data + 2);
-
-  if (m_thd.prevFrameSize)
+  if (m_packerMAT->HaveOutput())
   {
-    uint16_t deltaSamples = inputTiming - m_thd.prevFrameTime;
-    /*
-     * One multiple-of-48kHz frame is 1/1200 sec and the IEC 61937 rate
-     * is 768kHz = 768000*4 bytes/sec.
-     * The nominal space per frame is therefore
-     * (768000*4 bytes/sec) * (1/1200 sec) = 2560 bytes.
-     * For multiple-of-44.1kHz frames: 1/1102.5 sec, 705.6kHz, 2560 bytes.
-     *
-     * 2560 is divisible by samplesPerFrame.
-     */
-    int deltaBytes = deltaSamples * 2560 / m_thd.samplesPerFrame;
-
-    /* padding needed before this frame */
-    paddingRem = deltaBytes - m_thd.prevFrameSize;
-
-    // detects stream discontinuities
-    if (paddingRem < 0 || paddingRem >= MAT_FRAME_SIZE * 2)
-    {
-      m_thd = {}; // recovering after seek
-      return;
-    }
+    std::vector<uint8_t> mat{m_packerMAT->GetOutputFrame()};
```

Comment:
If this returns a default constructed std::vector can the next line function? ie. Is it valid to not return output frames?